metrics/prometheus.go

```diff
@@ -0,0 +1,155 @@
+package metrics
+
+import (
+	"net/http"
+	"os"
+	"sync"
+
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+const (
+	namespace = "traefik"
+)
+
+// TODO: #156 - More detailed metrics
+type entrypointMetrics struct {
+}
+
+// TODO: #156 - More detailed metrics
+type providerMetrics struct {
+}
+
+// TODO: #156 - More detailed metrics
+type frontendMetrics struct {
+}
+
+// TODO: #156 - More detailed metrics
+type backendMetrics struct {
+}
+
+type globalMetrics struct {
+	UpTime                    prometheus.Gauge
+	RequestResponseTimeTotal  prometheus.Gauge
```

Comment:
aha, I see. So conceptually this is a counter (because it only ever increases) but because we want to `set()` we need to use a gauge (for now). AFAIK there is nothing we miss out on by using gauges except maybe some clarity.

Throwing in all the labels is what I'm really excited about because that model is the perfect fit for how traefik operates imho :)